dashboard/src/utils/bumpDate.js

```diff
@@ -0,0 +1,6 @@
+const bumpToDate = (date, numberOfDays) => {
```

Comment:
This function is fine, but it should probably have a "docstring", "header comment", or whatever it's called in Javascript pointing out that it modifies its input.

I haven't worked with Javascript enough to get a feel for the "culture", but I get the impression that it might be more culturally compatible if this function returned a _new_ `Date` object with the requested value, rather than modifying the input one...so, if it's not going to do that, we should probably point it out!  I don't know that that would be more convenient for the way that this function is actually used by our code, but it probably deserves some consideration.

Reply:
JSDoc comments everywhere would make the code a lot easier to read; including a description of the basic operation, the meaning and type of each parameter, and the return type.

While I wouldn't ask Shubham to retrofit the entire codebase now, I agree that documentation of the new methods would be "highly desirable". And that's especially true if methods are doing unusual things like modifying the input parameters, which are not going to be obvious to anyone at first glance!

(And in this case, I hadn't thought about that aspect; but if that's what `date.setUTCDate` does, instead of returning a new `Date` object, you probably shouldn't be doing it that way.)

Reply:
Indeed:  [`Date.setUTCDate()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setUTCDate) (along with all of the other `set*()` functions) modify the specified object rather than generating new ones.